Source/AhriAndBear/Environments/ABRestArea.cpp

```diff
@@ -34,14 +38,25 @@ void AABRestArea::BeginOverlap(AActor* self, AActor* OtherActor)
 	auto animal = Cast<AABAnimalCharacter>(OtherActor);
 	if (animal == nullptr) return;
 
-	UABSurvivalStatFunctions::SetRateOfChange(animal->SurvivalComponent->Hunger, 1.f);
-	UABSurvivalStatFunctions::SetRateOfChange(animal->SurvivalComponent->Thirst, 1.f);
+	animal->SurvivalComponent->AddModifier(this);
 }
 
 void AABRestArea::EndOverlap(AActor* self, AActor* otherActor) {
+	GEngine->AddOnScreenDebugMessage(-1, 5.f, FColor::Red, TEXT("Overlap over"));
```

Comment:
Don't forget to remove the debug message in the build, I don't know why Unreal won't ignore them in the build